src/App.tsx

```diff
@@ -1,26 +1,347 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { addTodo, deleteTodo, getTodos, updateTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { USER_ID } from './constants';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoServiceErrors } from './types/TodoServiceErrors';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [todosErrorMessage, setTodosErrorMessage] =
+    useState<TodoServiceErrors | null>(null);
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(
+    FilterStatus.All,
+  );
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [togglingTodoIds, setTogglingTodoIds] = useState<number[]>([]);
+
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
```

Comment:
```suggestion
  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
```
do we really need this state? can we find todo id from todos array?

Reply:
At the moment, it is not possible to remove/replace this state, because the logic for displaying the edit form is tied to it.